app/src/main/java/org/kiwix/kiwixmobile/qr/ShareByQRCodeDialog.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2024 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.qr
+
+import android.app.AlertDialog
+import android.app.Dialog
+import android.os.Bundle
+import androidx.fragment.app.DialogFragment
+import androidx.navigation.fragment.navArgs
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.databinding.DialogShareByQrCodeBinding
+import org.kiwix.kiwixmobile.core.qr.GenerateQR
+
+class ShareByQRCodeDialog : DialogFragment() {
```

Comment:
Please move this class to the `core` module like the `GenerateQR` class. Also, rename this class to `QRCodeDialog` since this only shows the QR code and does not share anything, so `QRCodeDialog` would be better for this.

Reply:
@MohitMaliFtechiz is there documentation on the concerns for each module? 

I noticed `core` seemed to center around domain logic but also had Android views. Opted here to keep Android dependent code out of domain.

Happy to move, just hoping to understand intent of modules better.

Reply:
@shaunburch `core` module is the base for both app and custom apps. We put the common functionality in the `core` module to use it in both modules that's why you are seeing some Android views in that module. For the server functionality, we will enable this feature in custom apps that's why I am asking you to put this in the core module.